### PR TITLE
UIIN-1975: New Permission: View source (instance)

### PR DESCRIPTION
**Pull request by Svitlana-Korol**
## Description
Created a new permission called View source (instance) 

## Issues
[UIIN-1975](https://issues.folio.org/browse/UIIN-1975)

### CODE REVIEW

File: src/ViewInstance.js [hunk 1]
```diff
@@ -556,38 +558,42 @@ class ViewInstance extends React.Component {
...
+                  >
+                    <Icon icon="duplicate">
+                      <FormattedMessage id="ui-inventory.duplicateInstanceMarc" />
+                    </Icon>
+                  </Button>
+                </IfPermission>
+              </>
+              )}
+              {canViewMARC && (
               <IfPermission perm="ui-quick-marc.quick-marc-holdings-editor.create">
```
Comment (BogdanDenis) [inline src/ViewInstance.js:597]: This button is to create MARC Holdings, button for viewing source should be somewhere else

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/ViewInstance.js [hunk 2]
```diff
@@ -503,7 +504,7 @@ class ViewInstance extends React.Component {
           </IfInterface>
 
           {
-            canEditDeriveMARCRecord && (
+            (canEditDeriveMARCRecord || canViewMARCSource) && (
```
Comment (BogdanDenis) [inline src/ViewInstance.js:507]: We only need to check for `canViewMARCSource`. Permission to view MARC record is included in permissions to edit and derive, so there's no need to check twice here

File: src/ViewInstance.js [hunk 3]
```diff
@@ -556,38 +557,36 @@ class ViewInstance extends React.Component {
...
-                  }}
-                >
-                  <Icon icon="duplicate">
-                    <FormattedMessage id="ui-inventory.duplicateInstanceMarc" />
-                  </Icon>
-                </Button>
-              </IfPermission>
-
+              {canEditMARCRecord && (
+              <Button
```
Comment (BogdanDenis) [inline src/ViewInstance.js:561]: Please move this button one tab to the right

File: src/ViewInstance.js [hunk 4]
```diff
@@ -556,38 +557,36 @@ class ViewInstance extends React.Component {
...
+                  this.editInstanceMarc();
+                }}
+              >
+                <Icon icon="edit">
+                  <FormattedMessage id="ui-inventory.editInstanceMarc" />
+                </Icon>
+              </Button>
+              )}
+              {canDeriveMARCRecord && (
+              <Button
```
Comment (BogdanDenis) [inline src/ViewInstance.js:576]: Same here

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/ViewInstance.js (hunk 2 above)
Comment (Svitlana-Korol) [reply src/ViewInstance.js]: Thanks, you are right, I fixed it.

File: src/ViewInstance.js [hunk 5]
```diff
@@ -406,12 +406,12 @@ class ViewInstance extends React.Component {
...
-    const canEditMARCRecord = stripes.hasPerm('records-editor.records.item.put');
-    const canDeriveMARCRecord = stripes.hasPerm('records-editor.records.item.post');
+    const canEditMARCRecord = stripes.hasPerm('ui-quick-marc.quick-marc-editor.all');
+    const canDeriveMARCRecord = stripes.hasPerm('ui-quick-marc.quick-marc-editor.duplicate');
     const hasReorderPermissions = stripes.hasPerm('ui-requests.create') || stripes.hasPerm('ui-requests.edit') || stripes.hasPerm('ui-requests.all');
+    const canViewMARCSource = stripes.hasPerm('ui-quick-marc.quick-marc-editor.view');
 
-    const canCreateMARCHoldingsForInstanceWithSourceMARC = isSourceMARC && canCreateMARCHoldings;
-    const canEditDeriveMARCRecord = isSourceMARC && (canEditMARCRecord || canDeriveMARCRecord);
+    const showQuickMarcMenuSection = (isSourceMARC && canCreateMARCHoldings) || (isSourceMARC && (canEditMARCRecord || canDeriveMARCRecord));
```
Comment (BogdanDenis) [inline src/ViewInstance.js:414]: This can be simplified a bit: `isSourceMARC && (canCreateMARCHoldings || canEditMARCRecord || canDeriveMARCRecord)`

**Review by BogdanDenis — APPROVED** [review]

**Review by mariiaschv — APPROVED** [review]

**Pull request merged**